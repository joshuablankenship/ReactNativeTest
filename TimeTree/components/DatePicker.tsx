import React, {useState} from 'react';
import {View, Platform, Text, Modal, TouchableOpacity} from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';
import {styles} from '../styles/CreateProfileStyles';

const MyDatePicker = (props) => {
  const [date, setDate] = useState(new Date(1598051730000));
  const [dateFormatted, setDateFormatted] = useState('Not Set');
  const [mode, setMode] = useState('date');
  const [show, setShow] = useState(false);

  const onChange = (event, selectedDate) => {
    const currentDate = selectedDate || date;
    setShow(Platform.OS === 'ios');
    setDate(currentDate);
    formatDate(currentDate);
  };

  const showMode = currentMode => {
    setShow(true);
    setMode(currentMode);
  };
  const showFalse = currentMode => {
    setShow(false);
  };
  const showDatepicker = () => {
    formatDate(date);
    showMode('date');
  };
  const formatDate= (d) => {
    let ye = new Intl.DateTimeFormat('en', { year: 'numeric' }).format(d);
    let mo = new Intl.DateTimeFormat('en', { month: 'short' }).format(d);
    let da = new Intl.DateTimeFormat('en', { day: '2-digit' }).format(d);
    setDateFormatted(`${mo} ${da} ${ye}`);
  };
  return (
    <View>
      <View style={styles.textinput}>
        <Text style={styles.text} onPress={showDatepicker}>
          {dateFormatted}
        </Text>
      </View>
      <View>
        <Modal
          animationType="slide"
          transparent={true}
          visible={show}
          presentationStyle="overFullScreen">
          <View style={styles.modalBackground}>
            <View style={styles.modalForeground}>
              <View style={styles.modal}>
                <Text style={styles.textDate}>{dateFormatted}</Text>
              </View>
              <View style={styles.text} />
              <View style={styles.modalForeground}>
                <DateTimePicker
                  style={styles.dateStyle}
                  testID="dateTimePicker"
                  value={date}
                  mode={mode}
                  display="spinner"
                  onChange={onChange}
                />
                <View style={styles.modalButtons}>
                  <TouchableOpacity 
                    onPress={showFalse}
                  ><Text style={styles.modalButtonWhite}>Cancel</Text></TouchableOpacity>
                  <TouchableOpacity
                    onPress={showFalse}
                  ><Text style={styles.textGreen}>Confirm</Text></TouchableOpacity>
                </View>
              </View>
            </View>
          </View>
        </Modal>
      </View>
    </View>
  );
};

export default MyDatePicker;
