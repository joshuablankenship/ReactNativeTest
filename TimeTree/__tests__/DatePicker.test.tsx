import {shallow, ShallowWrapper} from 'enzyme';
import React from 'react';
import {TouchableOpacity} from 'react-native';
import DatePicker from '../components/DatePicker';
import DateTimePicker from '@react-native-community/datetimepicker';
const createTestProps = (props: any) => ({
  navigation: {
    navigate: jest.fn(),
  },
  closeLoginSheet: jest.fn(),
  ...props,
});
describe('Test DatePicker Component', () => {
  let wrapper: ShallowWrapper;
  let props: any;
  wrapper = shallow(<DatePicker {...props} />);
  props = createTestProps({
    navigate: () => {},
    closeLoginSheet: () => {},
  });
  it('It should match initial snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('It should display three touchable opacity components', () => {
    expect(wrapper.find(TouchableOpacity).length).toEqual(3);
  });
  it('It should contain a DateTimePicker', () => {
    expect(wrapper.find(DateTimePicker).length).toEqual(1);
  });
});
