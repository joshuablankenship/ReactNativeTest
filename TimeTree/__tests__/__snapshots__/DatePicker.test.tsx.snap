// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DatePicker Component It should match initial snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#f8f8f8",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#f8f8f8",
          "borderColor": "#f8f8f8",
          "borderWidth": 1,
          "color": "#8C8C8C",
          "flexDirection": "row",
          "fontWeight": "bold",
          "height": 53,
          "justifyContent": "space-between",
          "padding": 10,
          "paddingTop": 15,
          "textAlignVertical": "center",
        }
      }
    />
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/close-button-png.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 15,
            "justifyContent": "center",
            "marginRight": 15,
            "marginTop": 20,
            "width": 15,
          }
        }
      />
    </ForwardRef>
  </View>
  <View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      presentationStyle="overFullScreen"
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "flex": 1,
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 200,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "marginTop": "70%",
              }
            }
          >
            <Text
              style={
                Object {
                  "alignItems": "center",
                  "color": "#2FCC87",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "padding": 20,
                  "textAlign": "center",
                }
              }
            />
          </View>
          <View />
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Picker
              display="spinner"
              mode="date"
              onChange={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                }
              }
              testID="dateTimePicker"
              value={2020-08-21T23:15:30.000Z}
            />
            <View
              style={
                Object {
                  "backgroundColor": "#f8f8f8",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <ForwardRef
                onPress={[Function]}
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "#8C8C8C",
                      "fontSize": 20,
                      "fontWeight": "bold",
                      "padding": 10,
                    }
                  }
                >
                  Cancel
                </Text>
              </ForwardRef>
              <ForwardRef
                onPress={[Function]}
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "#2FCC87",
                      "fontSize": 20,
                      "fontWeight": "bold",
                      "padding": 10,
                    }
                  }
                >
                  Confirm
                </Text>
              </ForwardRef>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
